rust/operator-binary/src/main.rs

```diff
@@ -108,13 +115,16 @@ async fn main() -> anyhow::Result<()> {
             .run(
                 pod_driver_controller::reconcile,
                 pod_driver_controller::error_policy,
-                Arc::new(pod_driver_controller::Ctx {
-                    client: client.clone(),
-                }),
+                Arc::new(client.clone()),
             )
             .map(|res| report_controller_reconciled(&client, &format!("{OPERATOR_NAME}.{POD_DRIVER_CONTROLLER_NAME}"), &res))
             .instrument(info_span!("pod_driver_controller"));
 
+            // Create new object because Ctx cannot be cloned
```

Comment:
Why can't clone not be implemented?

Could there maybe be a function on the Ctx for a pseudo clone?

Reply:
It's an unfortunate omission that `ProductConfigManager` does not derive Clone and the `config` field is private. It should be fixed in operator-rs but I guess there was no need for it.